tree/dataframe/src/RLoopManager.cxx

```diff
@@ -369,20 +369,31 @@ RLoopManager::RLoopManager(std::unique_ptr<RDataSource> ds, const ColumnNames_t
 }
 
 RLoopManager::RLoopManager(ROOT::RDF::Experimental::RDatasetSpec &&spec)
-   : fBeginEntry(spec.GetEntryRangeBegin()),
-     fEndEntry(spec.GetEntryRangeEnd()),
-     fSamples(spec.MoveOutSamples()),
-     fNSlots(RDFInternal::GetNSlots()),
+   : fNSlots(RDFInternal::GetNSlots()),
      fLoopType(ROOT::IsImplicitMTEnabled() ? ELoopType::kROOTFilesMT : ELoopType::kROOTFiles),
      fNewSampleNotifier(fNSlots),
      fSampleInfos(fNSlots),
      fDatasetColumnReaders(fNSlots)
 {
+   ChangeSpec(std::move(spec));
+}
+
+void RLoopManager::ChangeSpec(ROOT::RDF::Experimental::RDatasetSpec &&spec)
```

Comment:
Indeed, the new spec is expected to provide exactly the same dataset schema. I agree with you that there is a very dangerous interaction potentially. We can't really do otherwise for now. I will add some very explicit docs in the next commit